File: solutions/lua/darts/1/darts.lua
local Darts = {}

function Darts.score(x, y)
  local dist = math.sqrt((x * x) + (y * y))
  if dist > 10 then return 0 end
  if dist <= 1 then return 10 end
  if dist <= 5 then return 5 end
  return 1
end

return Darts

File: solutions/ruby/darts/1/darts.rb
# frozen_string_literal: true

# class Darts
class Darts
  attr_reader :score

  def initialize(coord_x, coord_y)
    dist = Math.sqrt(coord_x * coord_x + coord_y * coord_y)
    @score = if dist <= 1 then 10
             elsif dist <= 5 then 5
             elsif dist <= 10 then 1
             else 0
             end
  end
end
